task/run-script-oci-ta/0.1/run-script-oci-ta.yaml

```diff
@@ -0,0 +1,249 @@
+      type: string
+    - name: SCRIPT_OUTPUT
+      description: The output string of the script
+      type: string
+    - name: SCRIPT_OUTPUT_ARRAY
+      description: An array of data generated by the script run. This can be used as input to future tasks' parameters.
+      type: array
+  volumes:
+    - name: trusted-ca
+      configMap:
+        items:
+          - key: $(params.caTrustConfigMapKey)
+            path: ca-bundle.crt
+        name: $(params.caTrustConfigMapName)
+        optional: true
+    - name: workdir
+      emptyDir: {}
+    - name: varlibcontainers
+      emptyDir: {}
+  stepTemplate:
+    computeResources:
+      limits:
+        memory: 4Gi
+      requests:
+        cpu: "1"
+        memory: 1Gi
+    env:
+      - name: SCRIPT
+        value: $(params.SCRIPT)
+      - name: HERMETIC
+        value: $(params.HERMETIC)
+      - name: SOURCE_CODE_DIR
+        value: source
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+    - name: use-trusted-artifact
+      image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:ff35e09ff5c89e54538b50abae241a765b2b7868f05d62c4835bebf0978f3659
+      args:
+        - use
+        - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+      volumeMounts:
+        - mountPath: /etc/pki/tls/certs/ca-custom-bundle.crt
+          name: trusted-ca
+          readOnly: true
+          subPath: ca-bundle.crt
+    - name: run-script
+      image: quay.io/konflux-ci/buildah-task:latest@sha256:ab0ba3b70f99faa74d2dd737422a965197af4922dec0109113bc535a94db0dfd
+      args:
+        - $(params.SCRIPT)
+      workingDir: /var/workdir
+      volumeMounts:
+        - mountPath: /mnt/trusted-ca
+          name: trusted-ca
+          readOnly: true
+        - mountPath: /var/lib/containers
+          name: varlibcontainers
+      env:
+        - name: SCRIPT_WORKDIR
+          value: $(params.SCRIPT_WORKDIR)
+        - name: SCRIPT_RUNNER_IMAGE
+          value: $(params.SCRIPT_RUNNER_IMAGE)
+        - name: STORAGE_DRIVER
+          value: $(params.STORAGE_DRIVER)
+      script: |
+        #!/bin/bash
+        set -euo pipefail
+        ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+        if [ -f "$ca_bundle" ]; then
+          echo "INFO: Using mounted CA bundle: $ca_bundle"
+          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          update-ca-trust
+        fi
+
+        # Fixing group permission on /var/lib/containers
+        chown root:root /var/lib/containers
+
+        sed -i 's/^\s*short-name-mode\s*=\s*.*/short-name-mode = "disabled"/' /etc/containers/registries.conf
+
+        # Setting new namespace to run buildah - 2^32-2
+        echo 'root:1:4294967294' | tee -a /etc/subuid >>/etc/subgid
+
+        UNSHARE_ARGS=()
+        if [ "${HERMETIC}" == "true" ]; then
+          echo "The script will be executed with network isolation"
+          UNSHARE_ARGS+=("--net")
+        fi
+
+        cp -r /var/workdir/source /var/workdir/script
```

Comment:
Do we need the `cp`? I think we can let the script modify `/var/workdir/source` directly

Reply:
Great question. I would say I had to do that on my first implementation attempt where the script_runner_image was the image task and I was not using `unshare` and `buildah` and did not try without it after moving to the current pattern. Will give a try and come back here with the results :)

Reply:
Tested and the copy is not required anymore. Moving to unshare + buildah to run the script fixed the original issue I had I guess.